eox_nelp/course_experience/api/v1/views.py

```diff
@@ -0,0 +1,193 @@
+
+from eox_nelp.course_experience.models import LikeDislikeCourse, LikeDislikeUnit, ReportCourse, ReportUnit
+
+from .serializers import (
+    LikeDislikeCourseExperienceSerializer,
+    LikeDislikeUnitExperienceSerializer,
+    ReportCourseExperienceSerializer,
+    ReportUnitExperienceSerializer,
+)
+from .validators import validate_course_key, validate_usage_key, validate_user
+
+
+class BaseJsonAPIView(ModelViewSet):
+    """class to configure base json api parameter
+
+    Args:
+        ModelViewSet : Django rest json api ModelViewSet
+    """
+    allowed_methods = ["POST", "GET", "PATCH"]
+
+    authentication_classes = (JwtAuthentication, SessionAuthenticationAllowInactiveUser)
+    permission_classes = (IsAuthenticated,)
+
+    pagination_class = JsonApiPageNumberPagination
+    parses_clasess = [
+        JSONParser,
+        FormParser,
+        MultiPartParser,
+    ]
+    renderer_classes = [JSONRenderer, BrowsableAPIRenderer] if getattr(settings, 'DEBUG', None) else [JSONRenderer]
+
+    metadata_class = JSONAPIMetadata
+    schema_class = AutoSchema
+    filter_backends = [
+        QueryParameterValidationFilter,
+        OrderingFilter,
+        DjangoFilterBackend,
+        SearchFilter,
+    ]
+    search_param = "filter[search]"
+
+
+class ExperienceView(BaseJsonAPIView):
+    """Class to set functionality of an ExperienceView.
+
+    Args:
+        BaseJsonAPIView: Inherited for the rest json api config.
+    """
+    def get_queryset(self, *args, **kwargs):
+        """Filter the queryset before being used.
+
+        Returns:
+            Queryset: queysyset using the super method, but filtered.
+        """
+        return super().get_queryset(*args, **kwargs).filter(author_id=self.request.user.id)
+
+    def create(self, request, *args, **kwargs):
+        """Perform processing for the request before use the base create method.
+        Args:
+            request: the request that arrives for create options.
+
+        Returns:
+            The return of ancestor create method with the request after processing.
+        """
+        if isinstance(request.data.get("course_id", {}), dict):
+            validate_course_key(request.data.get("course_id", {}).get("id", None))
+        validate_user(request.user)
+        request = self.change_author_data_2_request_user(request)
+        return super().create(request, *args, **kwargs)
+
+    def update(self, request, *args, **kwargs):
+        """Perform processing for the request before use the base update method.
+        Args:
+            request: the request that arrives for create options.
+
+        Returns:
+            The return of ancestor update method with the request after processing.
+        """
+        validate_user(request.user)
+        request = self.change_author_data_2_request_user(request)
+        return super().update(request, *args, **kwargs)
+
+    def change_author_data_2_request_user(self, request):
```

Comment:
I prefer this logic in the serializer, I think the serializer has a create method